### PR TITLE
Feature/GitHub action

### PR DESCRIPTION
**Pull request by ashraf-km**
### All Submissions:

* [ ] Have you followed the guidelines in our [Contributing](/contributing.md) document?
* [ ] Have you read and signed the automated Contributor's License Agreement?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### New Feature Submissions:

1. [ ] Does your submission pass tests?
2. [ ] Have you lint your code locally prior to submission?

### Changes to Core Features:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your core changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

### CODE REVIEW

**Review by javidlulu — COMMENTED** [review]
Check if the coveralls is working fine

File: src/afterSuccess.sh [hunk 1]
```diff
@@ -0,0 +1 @@
+bash <(curl -s https://copilot.blackducksoftware.com/ci/travis/scripts/upload)
```
Comment (shreyanshp) [inline src/afterSuccess.sh:1]: I think this URL will not be the same, read the documentation at https://copilot.blackducksoftware.com/

Comment (shreyanshp) [reply src/afterSuccess.sh:1]:
```
Setting up CoPilot requires modifying your workflow file (a file in .github/workflows/). Once you commit the change and your next build succeeds, you will find your results on this page! Please note that your build must be passing in order for CoPilot to generate results.

First, if you are not running in a Java environment, you'll have to setup Java:

- name: Set up Java (CoPilot)
      uses: actions/setup-java@v1
      with:
        java-version: 1.8
The following can then be added after your Java setup block to enable results:

- name: Upload to CoPilot
      if: github.event_name == 'push' || github.event_name == 'pull_request'
      run: bash <(curl -s https://copilot.blackducksoftware.com/ci/githubactions/scripts/upload)
One last thing – in order to to see which components were added or removed in pull requests, your CI workflow needs to be running builds including this step for both push and pull_request events.
```

Comment (shreyanshp) [reply src/afterSuccess.sh:1]: image

Comment (shreyanshp) [reply src/afterSuccess.sh:1]: https://github.com/BlackDuckCoPilot?utf8=%E2%9C%93&tab=repositories&q=githubactions

**Comment by shreyanshp** [comment]
@ashraf-km Also fix the security issue raised by BlackDuck which I DM'd you

**Comment by javidlulu** [comment]
Closing this, since coverall not working from fork, raising another PR from main repo

**Comment by shreyanshp** [comment]
@javidlulu That seems very odd that coveralls won't work from fork, do you have any evidence of actual issue?
We may get PRs from external forks too, how do you plan to check coverage from external fork's PRs?

**Pull request closed without merging**